client/src/main/java/com/vesoft/nebula/client/example/GraphClientExample.java

```diff
@@ -0,0 +1,102 @@
+
+    private static final String[] insertEdges = {
+            "INSERT EDGE select(grade) VALUES 200 -> 101:(5);",
+            "INSERT EDGE select(grade) VALUES 200 -> 102:(3);",
+            "INSERT EDGE select(grade) VALUES 201 -> 102:(3);",
+            "INSERT EDGE select(grade) VALUES 202 -> 102:(3);",
+            "INSERT EDGE like(likeness) VALUES 200 -> 201:(92.5);",
+            "INSERT EDGE like(likeness) VALUES 201 -> 200:(85.6);",
+            "INSERT EDGE like(likeness) VALUES 201 -> 202:(93.2);"
+    };
+
+    public static void main(String[] args) throws ConnectionException, TException, NGQLException {
+        GraphClient client = new GraphClientImpl("127.0.0.1", 3699);
+        client.connect("user", "password");
+        int code = client.execute(SPACE_NAME);
+        if (code != ErrorCode.SUCCEEDED) {
+            LOGGER.error(String.format("Switch Space %s Failed", SPACE_NAME));
+            LOGGER.error(String.format("Please confirm %s have been created", SPACE_NAME));
+            System.exit(-1);
+        }
+
+        for (String statement : insertVertices) {
+            code = client.execute(statement);
+            if (code != ErrorCode.SUCCEEDED) {
+                LOGGER.error(String.format("Insert Vertices Failed: %s", statement));
+                System.exit(-1);
+            }
+        }
+
+        for (String statement : insertEdges) {
+            code = client.execute(statement);
+            if (code != ErrorCode.SUCCEEDED) {
+                LOGGER.error(String.format("Insert Edges Failed: %s", statement));
+                System.exit(-1);
+            }
+        }
+
+        ResultSet resultSet = client.executeQuery("GO FROM 201 OVER like;");
+        List<String> columns = resultSet.getColumns()
+                                        .stream()
+                                        .map(bytes -> new String(bytes))
+                                        .collect(Collectors.toList());
+        LOGGER.info(String.format("Columns: %s", Joiner.on(" ").join(columns)));
+
+        for (RowValue value : resultSet.getRows()) {
+            LOGGER.info("ID: %d", value.columns.get(0).getId());
+        }
+
+        resultSet = client.executeQuery("GO FROM 201 OVER like " +
+                "WHERE $$.student.age >= 17 YIELD $$.student.name AS Friend, " +
+                "$$.student.age AS Age, $$.student.gender AS Gender;");
+
+        columns = resultSet.getColumns()
+                .stream()
+                .map(bytes -> new String(bytes))
+                .collect(Collectors.toList());
+
+        LOGGER.info(String.format("Columns: %s", Joiner.on(" ").join(columns)));
+
+        for (RowValue value : resultSet.getRows()) {
+            LOGGER.info("%s, %d, %s", new String(value.columns.get(0).getStr()),
+                                      value.columns.get(1).getInteger(),
+                                      new String(value.columns.get(2).getStr()));
+        }
+
+        client.disconnect();
```

Comment:
try... catch ... finally { client.disconnect(); }

Reply:
GraphClient extends AutoCloseable and implement close()